src/languages/dart.js

```diff
@@ -25,6 +25,15 @@ export default function(hljs) {
     keywords: 'true false null this is new super'
   };
 
+  const NUMBER = {
+    className: 'number',
+    relevance: 0,
+    variants: [
+      { begin: "\\b[0-9][0-9_]*(\\.[0-9][0-9_]*)?([eE][+-]?[0-9][0-9_]*)?\\b" },
+      { begin: "\\b0[xX][0-9A-Fa-f][0-9A-Fa-f_]*\\b" }
```

Comment:
Nice, but please use a regex literal, not a string with tons of extra escapes.  And reach for the new `match` vs the older `begin` (which is more for begin/end pairs).

Reply:
Thanks for the tip! Done